api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubV3.java

```diff
@@ -0,0 +1,196 @@
+
+        Map<String, Object> uniqueOptions = new HashMap<>();
+        uniqueOptions.put(REPO_URL, "");
+        uniqueOptions.put(BRANCH, "");
+        collector.setUniqueFields(uniqueOptions);
+
+        return collector;
+    }
+
+    public abstract String process(JSONObject jsonObject) throws MalformedURLException, HygieiaException, ParseException;
+    public abstract QueryDslPredicateExecutor<GitHubRepo> getGitHubRepoRepository();
+
+    protected CollectorItem buildCollectorItem (ObjectId collectorId, String repoUrl, String branch) {
+        if (!StringUtils.isEmpty(repoUrl) && !StringUtils.isEmpty(branch)) {
+            CollectorItem collectorItem = new CollectorItem();
+            collectorItem.setCollectorId(collectorId);
+            collectorItem.setEnabled(true);
+            collectorItem.setPushed(true);
+            collectorItem.setLastUpdated(System.currentTimeMillis());
+            collectorItem.getOptions().put(REPO_URL, repoUrl);
+            collectorItem.getOptions().put(BRANCH, branch);
+
+            return collectorItem;
+        }
+
+        return null;
+    }
+
+    protected CollectorItem getCollectorItem(String repoUrl, String branch) throws HygieiaException {
+        Collector col = collectorService.createCollector(getCollector());
+
+        if (col == null)
+            throw new HygieiaException("Failed creating collector.", HygieiaException.COLLECTOR_CREATE_ERROR);
+
+        CollectorItem item = buildCollectorItem(col.getId(), repoUrl, branch);
+        if (item == null)
+            throw new HygieiaException("Failed creating collector item. Invalid repo url and/or branch", HygieiaException.COLLECTOR_ITEM_CREATE_ERROR);
+
+        CollectorItem colItem = collectorService.createCollectorItem(item);
+
+        if (colItem == null)
+            throw new HygieiaException("Failed creating collector item.", HygieiaException.COLLECTOR_ITEM_CREATE_ERROR);
+
+        return colItem;
+    }
+
+    protected String getRepositoryToken(String scmUrl) {
+        Collector collector = collectorService.createCollector(getCollector());
+
+        Map<String, Object> options = new HashMap<>();
+        options.put(REPO_URL, scmUrl);
+
+        GitHubRepo gitHubRepo = findByCollectorIdAndOptions(collector.getId(), options);
+
+        if (gitHubRepo != null) {
+            return String.valueOf(gitHubRepo.getOptions().get(TOKEN));
+        }
+
+        return null;
+    }
+
+    protected GitHubRepo findByCollectorIdAndOptions( ObjectId collectorId, Map<String, Object> options) {
+        QCollectorItem item = new QCollectorItem("collectorItem");
+        BooleanBuilder builder = new BooleanBuilder();
+        builder.and(item.collectorId.eq(collectorId));
+        options.keySet().forEach(k -> {
+            builder.and(item.options.get(k).eq(options.get(k)));
+        });
+
+        Iterable<GitHubRepo> gitHubRepoIterable = getGitHubRepoRepository().findAll(builder.getValue());
```

Comment:
Should add this to CollectorItemRepository class

Reply:
Realized that there is an existing method in CollectorItemRepository that I could use, may be it was not there at the time I was building the logic or may be I missed to see it, I am not sure. But made changes to use the existing method, thanks